### PR TITLE
Expand OpenVPN data key usage

### PR DESCRIPTION
**Pull request by schwabe**

### CODE REVIEW

File: openvpn-wire-protocol.xml [hunk 1]
```diff
@@ -1533,15 +1541,95 @@ struct datakeys {
...
+        After a set time period the pending key is promoted to being the active key and
+        the active key is set to be the retired key (sometimes also referred to as lame
+        duck key). The retired key will be used only for decryption but no longer for
+        encryption.
+      </t>
+      <t>
+        In the default configuration of OpenVPN, the time period of promoting a key from
+        pending to active is calucated as
+        <artwork>
+          min(handshake_window, renegotiation_time/2)
```
Comment (reynir) [inline openvpn-wire-protocol.xml:1610]: What if `renegotiation_time` is 1? :D Is this then half a second? I ask because I use `--reneg-sec 1` in a test setup, and I am curious if sub second values are used.

Comment (schwabe) [reply openvpn-wire-protocol.xml:1610]: Then the implementation becomes broken and directly uses the new key since 1/2 = 0 in integer math:

From ssl.c:

```
static int
auth_deferred_expire_window(const struct tls_options *o)
{
    int ret = o->handshake_window;
    const int r2 = o->renegotiate_seconds / 2;

    if (o->renegotiate_seconds && r2 < ret)
    {
        ret = r2;
    }
    return ret;
}
```

I didn't want to include this that are more implementation bugs in corner cases into the RFC.

Comment (reynir) [reply openvpn-wire-protocol.xml:1610]: Ok, that makes sense. Thanks for the code snippet. I'll modify my test setup!
